lti_consumer/tests/unit/test_lti.py

```diff
@@ -0,0 +1,287 @@
+        u'''
+        {"@type": "Result",
+         "@context": "http://purl.imsglobal.org/ctx/lis/v2/Result",
+         "resultScore": {}}''',   # score TypeError
+    ], u"Could not convert resultScore to float"),
+]
+
+VALID_JSON_INPUTS = [
+    (u'''
+    {"@type": "Result",
+     "@context": "http://purl.imsglobal.org/ctx/lis/v2/Result",
+     "resultScore": 0.1}''', 0.1, u""),  # no comment means we expect ""
+    (u'''
+    [{"@type": "Result",
+     "@context": "http://purl.imsglobal.org/ctx/lis/v2/Result",
+     "@id": "anon_id:abcdef0123456789",
+     "resultScore": 0.1}]''', 0.1, u""),  # OK to have array of objects -- just take the first.  @id is okay too
+    (u'''
+    {"@type": "Result",
+     "@context": "http://purl.imsglobal.org/ctx/lis/v2/Result",
+     "resultScore": 0.1,
+     "comment": "ಠ益ಠ"}''', 0.1, u"ಠ益ಠ"),  # unicode comment
+    (u'''
+    {"@type": "Result",
+     "@context": "http://purl.imsglobal.org/ctx/lis/v2/Result"}''', None, u""),  # no score means we expect None
+]
+
+GET_RESULT_RESPONSE = {
+    "@context": "http://purl.imsglobal.org/ctx/lis/v2/Result",
+    "@type": "Result",
+}
+
+
+class TestParseResultJson(unittest.TestCase):
+    """
+    Unit tests for `lti_consumer.lti.parse_result_json`
+    """
+
+    def test_invalid_json(self):
+        """
+        Test invalid json raises exception
+        """
+        for error_inputs, error_message in INVALID_JSON_INPUTS:
+            for error_input in error_inputs:
+                with self.assertRaisesRegexp(LtiError, error_message):
+                    parse_result_json(error_input)
+
+    def test_valid_json(self):
+        """
+        Test valid json returns expected values
+        """
+        for json_str, expected_score, expected_comment in VALID_JSON_INPUTS:
+            score, comment = parse_result_json(json_str)
+            self.assertEqual(score, expected_score)
+            self.assertEqual(comment, expected_comment)
+
+
+class TestLti1Consumer(TestLtiConsumerXBlock):
+    """
+    Unit tests for Lti1Consumer
+    """
+
+    def setUp(self):
+        super(TestLti1Consumer, self).setUp()
+        self.lti_consumer = Lti1Consumer(self.xblock)
+
+    @patch('lti_consumer.lti.get_oauth_request_signature', Mock(
+        return_value=(
+            'OAuth oauth_nonce="80966668944732164491378916897", '
+            'oauth_timestamp="1378916897", oauth_version="1.0", oauth_signature_method="HMAC-SHA1", '
+            'oauth_consumer_key="t", oauth_signature="frVp4JuvT1mVXlxktiAUjQ7%2F1cw%3D"'
+        )
+    ))
+    @patch(
+        'lti_consumer.lti_consumer.LtiConsumerXBlock.prefixed_custom_parameters',
+        PropertyMock(return_value={u'custom_param_1': 'custom1', u'custom_param_2': 'custom2'})
+    )
+    @patch('lti_consumer.lti_consumer.LtiConsumerXBlock.lti_provider_key_secret', PropertyMock(return_value=('t', 's')))
+    @patch('lti_consumer.lti_consumer.LtiConsumerXBlock.user_id', PropertyMock(return_value='4knx2j'))
```

Comment:
Also same thing: this looks like a real value. You'll want to make it obviously-fake.